### PR TITLE
Verify the size of reference frames when saving them

### LINKED ISSUE #37: oss-fuzz reports an assert failure in libjxl
I think oss-fuzz has found an assert failure in libjxl. This file:

http://www.rollthepotato.net/~john/clusterfuzz-testcase-minimized-pngsave_buffer_fuzzer-6695474309496832.fuzz

Triggers this:

```
&nbsp; | /src/libjxl/lib/jxl/image_ops.h:25: JXL_ASSERT: SameSize(from, *to)
&nbsp; | AddressSanitizer:DEADLYSIGNAL
&nbsp; | =================================================================
&nbsp; | ==484==ERROR: AddressSanitizer: ILL on unknown address 0x00000250cd19 (pc 0x00000250cd19 bp 0x7f7603b124d0 sp 0x7f7603b124d0 T4)
&nbsp; | #0 0x250cd19 in jxl::Abort() libjxl/lib/jxl/base/status.cc:42:3
&nbsp; | #1 0x28dd2ff in CopyImageTo&lt;float&gt; libjxl/lib/jxl/image_ops.h:25:3
&nbsp; | #2 0x28dd2ff in jxl::ImageBlender::PrepareBlending(jxl::PassesDecoderState*, jxl::FrameOrigin, unsigned long, unsigned long, jxl::ColorEncoding const&amp;, jxl::ImageBundle*) libjxl/lib/jxl/blending.cc:130:9
&nbsp; | #3 0x2d5931d in jxl::FinalizeFrameDecoding(jxl::ImageBundle*, jxl::PassesDecoderState*, jxl::ThreadPool*, bool, bool) libjxl/lib/jxl/dec_reconstruct.cc:1081:5
&nbsp; | #4 0x2bc9b4a in jxl::FrameDecoder::Flush() libjxl/lib/jxl/dec_frame.cc:817:3
&nbsp; | #5 0x2bbe2b2 in jxl::FrameDecoder::FinalizeFrame() libjxl/lib/jxl/dec_frame.cc:849:3
&nbsp; | #6 0x2537b44 in jxl::(anonymous namespace)::JxlDecoderProcessInternal(JxlDecoderStruct*, unsigned char const*, unsigned long) libjxl/lib/jxl/decode.cc:1155:30
&nbsp; | #7 0x2532671 in JxlDecoderProcessInput libjxl/lib/jxl/decode.cc:1668:14
...
```

With this version of libjxl: https://gitlab.com/wg1/jpeg-xl/-/compare/040eae8105b61b312a67791213091103f4c0d034...30ea86ab4c1f1b98c21967a2e3d72a51fe77e454

### PR DESCRIPTION
**Pull request by sboukortt**
This should fix #37.

### CODE REVIEW

**Review by jonsneyers — APPROVED** [review]

**Comment by veluca93** [comment]
Uh, when can this happen? save_before_color_transform frames?

**Review by veluca93 — APPROVED** [review]

**Comment by sboukortt** [comment]
> Uh, when can this happen? save_before_color_transform frames?

As far as I can tell, at least in the file attached to #37, it’s a frame that `CanBeReferenced()` but which has a frame header indicating a different size than a subsequent frame.

**Comment by sboukortt** [comment]
In the end, per a discussion with @veluca93, this morphed into something rather different.

File: lib/jxl/blending.cc [hunk 1]
```diff
@@ -88,11 +90,12 @@ Status ImageBlender::PrepareBlending(PassesDecoderState* dec_state,
         info_.source);
   }
 
-  if (bg.xsize() != image_xsize || bg.ysize() != image_ysize ||
+  if (bg.xsize() < image_xsize || bg.ysize() < image_ysize ||
       bg.origin.x0 != 0 || bg.origin.y0 != 0) {
     return JXL_FAILURE("Trying to use a %zux%zu crop as a background",
                        bg.xsize(), bg.ysize());
   }
+  bg.ShrinkTo(image_xsize, image_ysize);
```
Comment (veluca93) [inline lib/jxl/blending.cc:98]: Do we need this line?

**Review by veluca93 — COMMENTED** [review]
Can we also add a check in frame_header.cc? (maybe it's not necessary)

File: lib/jxl/blending.cc (hunk 1 above)
Comment (sboukortt) [reply lib/jxl/blending.cc]: I guess not.

**Review by veluca93 — APPROVED** [review]

**Pull request merged**